go panic on committing

Hi,

It was first time to use git-lfs and the very first experiment was not so good. :weary:
On the first commit I faced the following panic stacktrace.

```
panic: runtime error: makeslice: len out of range

goroutine 68 [running]:
strings.Repeat(0x4fced0, 0x1, 0xffffffffffffffd9, 0x0, 0x0)
    /usr/local/Cellar/go/1.5.1/libexec/src/strings/strings.go:464 +0x5f
github.com/github/git-lfs/lfs.(*ProgressMeter).update(0xc820264000)
    /Users/rick/go/src/github.com/github/git-lfs/lfs/progress_meter.go:140 +0x7c1
github.com/github/git-lfs/lfs.(*ProgressMeter).writer(0xc820264000)
    /Users/rick/go/src/github.com/github/git-lfs/lfs/progress_meter.go:106 +0x30
created by github.com/github/git-lfs/lfs.(*ProgressMeter).Start
    /Users/rick/go/src/github.com/github/git-lfs/lfs/progress_meter.go:57 +0x60
```

Then I checked the code and the cause.
```go
// around lfs/progress_meter.go:140
func (p *ProgressMeter) update() {
	if p.dryRun || p.estimatedFiles == 0 {
		return
	}

	width := 80 // default to 80 chars wide if ts.GetSize() fails
	size, err := ts.GetSize()
	if err == nil {
		width = size.Col()
	}

	// (%d of %d files, %d skipped) %f B / %f B, %f B skipped
	// skipped counts only show when > 0

	out := fmt.Sprintf("\rGit LFS: (%d of %d files", p.finishedFiles, p.estimatedFiles)
	if p.skippedFiles > 0 {
		out += fmt.Sprintf(", %d skipped", p.skippedFiles)
	}
	out += fmt.Sprintf(") %s / %s", formatBytes(p.currentBytes), formatBytes(p.estimatedBytes))
	if p.skippedBytes > 0 {
		out += fmt.Sprintf(", %s skipped", formatBytes(p.skippedBytes))
	}

	padding := strings.Repeat(" ", width-len(out))
	fmt.Fprintf(os.Stdout, out+padding)
}
```

In my case, at the secondary last line `width-len(out)` was `-39`; enough to cause Go panic.
By tweaking like
```go
    padlen := width - len(out)
    if 0 < padlen {
        padding := strings.Repeat(" ", padlen)
        out += padding
    }
    fmt.Fprintf(os.Stdout, out)
```
then my first commit had been finished successfully.

I am not sure why `width` was `0`.
I use Emacs+magit or Emacs+shell on Mac.
Maybe `ts.GetSize()` needs to be improved. But still it is good to have guard code like above.

Anyway, thanks for providing `git-lfs`. I think I will love the tool a lot.:smiley:

--- a/lfs/progress_meter.go
+++ b/lfs/progress_meter.go
@@ -139,7 +139,7 @@ func (p *ProgressMeter) update() {
 
 	padlen := width - len(out)
 	if 0 < padlen {
-			out += strings.Repeat(" ", padlen)
+		out += strings.Repeat(" ", padlen)
 	}
 
 	fmt.Fprintf(os.Stdout, out)

--- a/lfs/progress_meter.go
+++ b/lfs/progress_meter.go
@@ -137,8 +137,12 @@ func (p *ProgressMeter) update() {
 		out += fmt.Sprintf(", %s skipped", formatBytes(p.skippedBytes))
 	}
 
-	padding := strings.Repeat(" ", width-len(out))
-	fmt.Fprintf(os.Stdout, out+padding)
+	padlen := width - len(out)
+	if 0 < padlen {
+			out += strings.Repeat(" ", padlen)
+	}
+
+	fmt.Fprintf(os.Stdout, out)
 }
 
 // progressLogger provides a wrapper around an os.File that can either

